lib/ecto/query/builder/select.ex

```diff
@@ -140,14 +140,18 @@ defmodule Ecto.Query.Builder.Select do
     end
   end
 
-  defp take?(fields) do
-    is_list(fields) and Enum.all?(fields, fn
-      {k, v} when is_atom(k) -> take?(List.wrap(v))
-      k when is_atom(k) -> true
-      _ -> false
+  defp take?(boolean) when is_boolean(boolean), do: false
```

Comment:
Interesting... what is the reason to avoid booleans?

Reply:
I was thinking of cases like `[:foo, [true: false]]` being passed 🤔.

Reply:
I mean... if they have a field named true, there is nothing we can do. :)

Reply:
that's true :D. I'll remove this check then.

Reply:
oh but then there exists this: https://github.com/elixir-ecto/ecto/blob/87ce8e3223a1c8a92d40536cf56ac83c0270d711/lib/ecto/changeset.ex#L1857

However, I could not find where `select(true)` is documented 🤔.

Reply:
I changed that to `select(1)` like what we did on `Repo.exists?`

Reply:
@qcam ah, but that may be a breaking change... I think we finally understand why we expect it to be a list then. With this new information, I think it may be better to improve the error message after all. WDYT?

Reply:
Yeah, maybe someone already depends on this behavior. Shall we document this?

I think currently if we pass `select(:foo)`, the error message is:

```
** (Ecto.Query.CompileError) `:foo` is not a valid query expression
```

I think that's reasonably clear. How do you think we would improve the message otherwise?

Reply:
It would be nice if we could say how to fix it. Maybe we could detect if something is atom and not nil/boolean, and say: "`:foo` is not a valid query expression. `:select` expects a query expression or a list of fields"?

Reply:
👍 I changed the implementation.